services/app-api/handlers/reports/fetch.ts

```diff
@@ -42,35 +42,29 @@ export const fetchReportsByState = handler(async (event, _context) => {
   };
 
   let startingKey;
-  let keepSearching = true;
   let existingItems = [];
   let results;
-  let count = 0;
 
-  const queryTable = async (keepSearching: boolean, startingKey?: any) => {
+  const queryTable = async (startingKey?: any) => {
     queryParams.ExclusiveStartKey = startingKey;
     let results = await dynamoDb.query(queryParams);
     if (results.LastEvaluatedKey) {
       startingKey = results.LastEvaluatedKey;
-      return [startingKey, keepSearching, results];
+      return [startingKey, results];
     } else {
-      keepSearching = false;
-      return [null, keepSearching, results];
+      return [null, results];
     }
   };
 
   // Looping to perform complete scan of tables due to 1 mb limit per iteration
-  while (keepSearching) {
-    count++;
-    if (count > 4) {
-      break;
-    }
-    [startingKey, keepSearching, results] = await queryTable(
-      keepSearching,
-      startingKey
+  do {
+    [startingKey, results] = await queryTable(startingKey);
+    existingItems.push(
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      ...results.Items.map(({ formTemplate, fieldData, ...item }) => item)
     );
-    existingItems.push(...results.Items);
-  }
+  } while (startingKey);
```

Comment:
Love the rewrite here, though I'm not quite following the map logic. How is it filtering?

Reply:
So in Javascript you can call out the fields you want to edit and overwrite them in. So you've likely seen something along the lines of this:

```
const object1 = {a: "value for a", b: "value for b"};

const object2 = {...object1, a: "updated value"};
```

object2 would then be `{a: "updated value", b: "value for b"};`


If we do a map in a similar fashion, we can call out the properties of the object we're mapping and just simply ignore them because we didn't return it. So using the spread operator I'm calling out all properties that aren't formTemplate and fieldData and returning that as I iterate through the array. It's a quick way to filter out properties that you don't want. Only downside is it leaves you with the unused vars scenario.

Reply:
Updated this to make it a lot more clear about whats happening. Totally not because it wouldn't deploy due to typescript linting errors